cudaaligner/src/min3.hpp

```diff
@@ -0,0 +1,25 @@
+/*
+* Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#pragma once
+
+namespace claragenomics
+{
+
+namespace cudaaligner
+{
+template <typename T>
+inline T const& min3(T const& t1, T const& t2, T const& t3)
```

Comment:
That's certainly a possibility. I'm not sure though, because this function is unlikely to be used anywhere else than in the cudaaligner algorithms. Either way works for me.

Reply:
I was thinking we might use it in `cudapoa` since we have similar `min` operations there as well. Don't remember if it exactly fits the use case, but let's just move it to common so it's there for others to use it :)

Reply:
Ok, then it makes sense. Will do that tomorrow.